python/ray/tune/search/variant_generator.py

```diff
@@ -425,8 +425,12 @@ def increment(i):
         yield unresolved_spec
         return
 
+    # a skeleton is easier to copy for every iteration
+    unresolved_spec_skeleton = copy.deepcopy(unresolved_spec)
+    for path, _ in grid_vars:
+        assign_value(unresolved_spec_skeleton, path, None)
     while value_indices[-1] < len(grid_vars[-1][1]):
-        spec = copy.deepcopy(unresolved_spec)
+        spec = copy.deepcopy(unresolved_spec_skeleton)
```

Comment:
Could you clarify the comments here to describe what's happening in each step? Specifically I think it would be good to explain what the "skeleton" is  (i.e. the spec with the `grid_vars` set to none), and then within the `while` loop comment that we are populating the values for a single variant.

Reply:
That's a great suggestion; I'll add the comments ASAP.

Reply:
Let me know if that's what you had in mind or if you have any feedback.